feat: 반납

lhj0621 opened the pull request:
## 작업 분류

- [ ] 버그 수정
- [x] 신규 기능 추가
- [ ] 리팩터링
- [ ] 테스트

## 작업 개요
-  반납 개발 

## 상세 내용
- 기본요금 계산
- 환승 처리
- 1분이내 처리
- 데이터 저장
- 이벤트 서비스 전달 데이터 가공
- findOne 에러 처리
- 이미 반납된 요청 에러 처리

## 집중 요망!
- 이벤트 측 개발 완료 후 주석 제거 필요

resolves: #11

gibson-lee93 committed:
feat: 반납 기본 요금 계산

- 기본요금 계산
- 환승 처리
- 1분이내 처리
- 데이터 저장
resolves: #11

lhj0621 committed:
feat : 반납 개발

- 반납 API 개발
- 이벤트 서비스 전달 데이터 가공
- findOne 에러 처리
- 이미 반납된 요청 에러 처리

resolves: #11

BangleCoding commented on src/history/history.repository.ts:
```diff
@@ -1,5 +1,123 @@
+    if (this.checkTransfer(user, history)) {
+      price = min_price * useMin;
+    }
+
```
사용시간 1분 미만인데 동시에 환승 30분 이내일 때에 해당되어서  환승 if문을 타버리는 경우는 없을까요?

chinsanchung replied:
조건문을 수정했습니다. 1분 미만일 때는 history 의 price 를 0으로 해서 업데이트하고 끝내도록 했습니다.

chinsanchung reviewed (commented):
참고: "group 으로 묶은 값을 리턴하기 전에 숫자형으로 바꿀 필요가 있음" => 수정 완료했습니다.

chinsanchung committed:
fix: 사용시간 1분에 대한 조건문 수정 및 sendData 수정

- 사용시간이 1분 미만일 경우 가격을 0으로 하여 내역을 갱신하고 끝내도록 설정
- sendData 의 isInArea, isInParkingzone, isInForbiddenArea 항목을 문자열에서 숫자로 수정

resolves: #25

chinsanchung committed:
fix: 리턴 값 수정

- 마지막 리턴을 sendData 에서 history 로 수정

resolves: #25

chinsanchung committed:
fix: saveHistory 메소드 작성 및 적용

- 반납하기 전 마지막으로 history 를 갱신하는 과정을 따로 메소드로 분리

resolves: #25
